### PR TITLE
DOCSP-25803 add configuration linking

### PR DESCRIPTION
**Pull request by davemungo**
[STAGING](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCSP-25803-improve-config-informatio-v1.0/reference/configuration/)

[JIRA](https://jira.mongodb.org/browse/DOCSP-25803)

[BUILD](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=63b6fb13dc742b91f2045256)

### CODE REVIEW

File: source/reference/configuration.txt [hunk 1]
```diff
@@ -12,16 +12,21 @@ Configuration
...
+The ``start`` command provides additional options for features like
+``reversibility`` and ``user write blocking``. For more details, see the 
+:ref:`start <c2c-api-start>` API documentation.
 
 Configuration File
 ==================
 
-Most :program:`mongosync` command-line options can be written to a YAML file.
-The configuration file specifies values for each setting using YAML format.
+Most :program:`mongosync` command-line options can be written to a YAML
```
Comment (davemungo) [inline source/reference/configuration.txt:26]: Unrelated fix - line length

File: source/reference/configuration.txt [hunk 2]
```diff
@@ -12,16 +12,21 @@ Configuration
    :depth: 2
    :class: singlecol
 
-You can configure :program:`mongosync` instances at startup using a configuration
-file.  The configuration file contains settings that are the equivalent of
-``mongosync`` command-line options.
+You can configure :program:`mongosync` instances at startup using a
```
Comment (davemungo) [inline source/reference/configuration.txt:15]: Unrelated fix - line length

File: source/reference/configuration.txt [hunk 3]
```diff
@@ -12,16 +12,21 @@ Configuration
...
+configuration file.  The configuration file contains settings that are
+the equivalent of ``mongosync`` command-line options.
 
+The ``start`` command provides additional options for features like
```
Comment (davemungo) [inline source/reference/configuration.txt:19]: The update. 

I swapped the usual link, monospace order since the distance between the references  is short and 'See also' needs to have a link.

File: source/reference/configuration.txt [hunk 4]
```diff
@@ -31,8 +36,8 @@ For example:
    logPath: "/var/log/mongosync"
    verbosity: "WARN"
 
-To use a configuration file, start ``mongosync`` with the :option:`--config`
-option to set the path to the file:
+To use a configuration file, start ``mongosync`` with the
```
Comment (davemungo) [inline source/reference/configuration.txt:39]: Unrelated fix - line length

File: source/reference/configuration.txt (hunk 3 above)
Comment (jeff-allen-mongo) [inline source/reference/configuration.txt:19]: [optional consideration]

You could maybe make this a note at the top of the Options section along the lines of:

"Not all options can be set in the configuration file. The ``start`` command provides additional options for features like..."

This might make it a bit clearer that you may need to use both the config file and API options.

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM % a consideration. Thanks for the update.

**Comment by davemungo** [comment]
@alan-zheng-mdb  this update adds a note based on user feedback from the docs. Would you have a look?

**Review by alan-zheng-mdb — APPROVED** [review]
LGTM
